### DISCUSSION TITLE
Event listener for Full screen view

### DISCUSSION THREAD

**Discussion by abhishekmardiya** [visgl/react-google-maps#156] [General]
Is there a way to listen for an event when a user clicks on the **Full Screen View** (toggle full screen view) ?

**Comment by usefulthink** [comment]
The Google Maps API doesn't communicate this, but it should be possible to work with the [`Document::fullscreenchange`](https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenchange_event) event to do what you need. Something like

```tsx
const [isFullscreen, setFullscreen] = useState(false);
useEffect(() => {
  document.addEventListener(
    'fullscreenchange', 
    (ev) => setFullscreen(Boolean(document.fullscreenElement))
  );
}, []);
```

**Reply by abhishekmardiya** [reply]
thank you !
